pydoop/hdfs/common.py

```diff
@@ -98,13 +116,30 @@ def __init__(self, m=None):
                 self.__error(m)
             else:
                 try:
-                    self.__binary = m[1] == "b"
+                    self.__text = m[1] == "t"
```

Comment:
This is slightly different from what io.open will do:  apparently, python3 does not differentiate between io.open(path, 'rb') and io.open(path, 'br'). I guess it is an undocumented feature of python3.

Reply:
Actually I've just discovered that `io.open` does accept `br` in Python2 (not `open` though). I chose the most restrictive behavior to make the implementation simpler, but we can extend this in the future.